### ISSUE TITLE
Same output for every input

### ISSUE THREAD

**Issue by Thrasher091** [deeplearning4j/deeplearning4j#7056]
#### Issue Description
Hi there!

I use a Neural Network for Deep Q Learning. After training it gives me the same outputs for every input.

My Input is an array with a size of 72 in which are either a one or a zero . I'm using backpropagation to train the Network.

I don't really know what I have to change, because I'm knew to dl4j and would appreciate some help.

Here is the code of my Network:

	
`
	
	private static MultiLayerNetwork MLN;
	int[] hiddenLayers =new int[]{54,36,18};
	public NeuralNetwork(){};

	public void createNetwork(){

		MultiLayerConfiguration configuration = null;

		NeuralNetConfiguration.ListBuilder builder = new NeuralNetConfiguration.Builder()
				
				
	
			
                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                .biasInit(0)
                .updater(new RmsProp.Builder().learningRate(0.001).build())
                .biasUpdater(new RmsProp.Builder().learningRate(0.001).build())
         
                .list()

                .layer(0, new DenseLayer.Builder()
                		.nIn(inputs)
                		.nOut(hiddenLayers[0])
                		.name("Input layer")
                        .weightInit(WeightInit.XAVIER)
                        .activation(Activation.RELU)
                        .name("Input Layer")
                        .build());
		
				builder.layer(1, new DenseLayer.Builder()
						.nIn(hiddenLayers[0])
						.nOut(hiddenLayers[1])
						.weightInit(WeightInit.XAVIER)
						.activation(Activation.RELU)
						.name("First hidden layer")
						.build());
				builder.layer(2, new DenseLayer.Builder()
						.nIn(hiddenLayers[1])
						.nOut(hiddenLayers[2])
						.weightInit(WeightInit.XAVIER)
						.activation(Activation.RELU)
						.name("Second hidden layer")
						.build());
				
				configuration =  builder.layer(2, new OutputLayer.Builder(LossFunction.MSE)
						.nIn(hiddenLayers[1])
						.nOut(outputs)
						.weightInit(WeightInit.XAVIER)
						.activation(Activation.RELU)
						.name("Output layer")
						.build())
						
							
				.pretrain(false)
				.backprop(true)
				.build();

	MultiLayerNetwork network = new MultiLayerNetwork(configuration);

	network.init();`

**Comment by AlexDBlack** [comment]
the network configuration looks ok at first glance
there's a bunch of questions though before we can start to debug this:
* What version of DL4J/RL4J are you using?
* How exactly are you concluding that it's giving you the same output? (note INDArray.toString reports limited number of digits)
* Have you tuned the network (learning rate) at all?

**Comment by Thrasher091** [comment]
Thanks for the answer @AlexDBlack !

1. I'm using dl4j 1.0.0-beta3. 

2. I pass various states through the network(with output(input)) and the outputs (Q-values for actions) are the same or nearly the same for all states.
When I play a little bit with the configuration the outputs become a little bit diffrent, but then they become nearly the same, a little bit like an average( e.g. 17.67  17.8 18.0 17.5 ...)

3. What do you mean with tuned? I only tried various learning rates but couldn't see a big difference.

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
